src/main/java/org/apache/spark/util/collection/unsafe/sort/PMemWriter.java

```diff
@@ -184,7 +184,7 @@ public void updateLongArray(LongArray sortedArray, int numRecords, int position)
             // update recordPointer in this array
             long originalRecordPointer = sortedArray.get(position);
             MemoryBlock page = taskMemoryManager.getOriginalPage(originalRecordPointer);
-            long baseOffset = page == null? 0: page.getBaseOffset();
+            long baseOffset = page.getBaseOffset();
```

Comment:
Should be 
```
(page == null)? 0: page.getBaseOffset()
```
?

Reply:
No. for offheap page, the pageObject is null but not page itself is null.

Reply:
ok